File: HW10/filter_plot.m
function filter_plot(s)
if ~isempty(instrfind)
    fclose(instrfind);
    delete(instrfind);
end

mySerial = serial(s,'BaudRate',9600,'Timeout',10);
fopen(mySerial);

fprintf('Port is open.')                               

% send 'r' to start reading
fprintf('r');

% store readings
n = 100;
data = zeros(n,4);
for i=1:n
    tmp = fscanf(mySerial, '%d %f %f %f');
    data(i,:) = str2num(tmp);
end

% parse readings
index = data(:,1);
maf = data(:,2);
iir = data(:,3);
fir = data(:,4);

% plot results
figure(1)
hold on
plot(index, maf, 'b')
plot(index, iir, 'r')
plot(index, fir, 'g')
title('Low Pass Filter Results')
legend('MAF','IIR','FIR')

end

File: HW10/filter_plot.py
#!/usr/bin/env python

import serial
import matplotlib.pyplot as plt

def main():
    port = serial.Serial('/dev/ttyACM0',9600,timeout=60)
    port.write('r')

    raw = []
    maf = []
    iir = []
    fir = []

    n = 100
    for i in range(n):
        read = port.readline()
        data = read.split() #Split string
        raw.append(float(data[1]))
        maf.append(float(data[2]))
        iir.append(float(data[3]))
        fir.append(float(data[4]))

    port.close()

    plt.figure(1)
    plt.plot(raw, label='Raw', color='k')
    plt.plot(maf, label='MAF', color='r')
    plt.plot(iir, label='IIR', color='g')
    plt.plot(fir, label='FIR', color='b')
    plt.legend()
    plt.show()

if __name__ == '__main__':
    main()
